src/adapters/fuel_adapter/fuel_client.rs

```diff
@@ -61,14 +61,26 @@ impl FuelAdapter for FuelClient {
             .await
             .map_err(|e| Error::NetworkError(e.to_string()))?;
 
-        Ok(maybe_block.map(Into::into))
+        match maybe_block {
```

Comment:
It would be better to adapt the conversion function to reduce duplication. 

Changes needed should look something like this:
```rust
use fuel_core_client::client::{types::block::Block as FuelGqlBlock, FuelClient as FuelGqlClient};

impl From<FuelGqlBlock> for FuelBlock {
    fn from(value: FuelGqlBlock) -> Self {
        FuelBlock {
            hash: *value.id,
            height: value.header.height,
        }
    }
}
```

Then the `maybe_block.map(Into::into)` will work again due to the conversion fn being adapted to the new block type that `0.19` uses.